org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/quickfix/TestLanguageQuickFixProvider.java

```diff
@@ -17,12 +20,26 @@
 
 public class TestLanguageQuickFixProvider extends AbstractDeclarativeIdeQuickfixProvider {
 
+	public static String EMF_QF_LABEL = "Change element name to first upper using object modification";
+	public static String TEXT_QF_LABEL = "Change element name to first upper using text replacement";
+
+	private String fixedName(TypeDeclaration declaration) {
+		return StringExtensions.toFirstUpper(declaration.getName());
+	}
+
 	@QuickFix(TestLanguageValidator.INVALID_NAME)
 	public void fixLowerCaseName(DiagnosticResolutionAcceptor acceptor) {
-		acceptor.accept("Change element name to first upper", obj -> {
-			final TypeDeclaration element = (TypeDeclaration) obj;
-			element.setName(StringExtensions.toFirstUpper(element.getName()));
+		acceptor.accept(EMF_QF_LABEL, obj -> {
+			final TypeDeclaration declaration = (TypeDeclaration) obj;
```

Comment:
Probably also from x2j, no need for the final

Reply:
done